src/generators/rust/RustGenerator.ts

```diff
@@ -0,0 +1,208 @@
+import { RustPreset, RUST_DEFAULT_PRESET } from './RustPreset';
+import { StructRenderer } from './renderers/StructRenderer';
+import { EnumRenderer } from './renderers/EnumRenderer';
+import { Logger } from '../../utils/LoggingInterface';
+import { RustDefaultConstraints, RustDefaultTypeMapping } from './RustConstrainer';
+import { TupleRenderer } from './renderers/TupleRenderer';
+import { UnionRenderer } from './renderers/UnionRenderer';
+import { PackageRenderer } from './renderers/PackageRenderer';
+
+export interface RustOptions extends CommonGeneratorOptions<RustPreset> {
+  typeMapping: TypeMapping<RustOptions>;
+  constraints: Constraints
+}
+
+export enum RustPackageFeatures {
+  json,
+}
+
+export interface RustPackageOptions {
+  packageName: string;
+  packageVersion: string;
+  authors: string[];
+  homepage: string;
+  repository: string;
+  license: string;
+  description: string;
+  edition: string;
+  packageFeatures: RustPackageFeatures[]
+}
+
+export interface RustRenderCompleteModelOptions {
+  supportFiles: boolean
+  package: RustPackageOptions
+}
+
+export const defaultRustRenderCompleteModelOptions = {
+  supportFiles: true,
+  package: {
+    // Many of these options can be parsed from schema?
+    packageName: 'asyncapi-rs-models',
+    packageVersion: '0.0.0',
+    authors: ['AsyncAPI Rust Champions'],
+    homepage: 'https://www.asyncapi.com/tools/modelina',
+    repository: 'https://github.com/asyncapi/modelina',
+    license: 'Apache-2.0',
+    description: 'Rust models generated by AsyncAPI Modelina',
+    edition: '2018',
+    packageFeatures: [RustPackageFeatures.json] as RustPackageFeatures[]
+  } as RustPackageOptions
+
+};
+
+export class ConstrainedSupportFileModel extends ConstrainedMetaModel { }
+
+/**
+ * Generator for Rust
+ */
+export class RustGenerator extends AbstractGenerator<RustOptions, RustRenderCompleteModelOptions> {
+  static defaultOptions: RustOptions = {
+    ...defaultGeneratorOptions,
+    defaultPreset: RUST_DEFAULT_PRESET,
+    typeMapping: RustDefaultTypeMapping,
+    constraints: RustDefaultConstraints,
+    indentation: {
+      type: IndentationTypes.SPACES,
+      size: 4
+    }
+  };
+  constructor(
+    options: Partial<RustOptions> = RustGenerator.defaultOptions,
+  ) {
+    const realizedOptions = { ...RustGenerator.defaultOptions, ...options };
+
+    super('Rust', realizedOptions);
+  }
+
+  splitMetaModel(model: MetaModel): MetaModel[] {
+    //These are the models that we have separate renderers for
+    const metaModelsToSplit = {
+      splitEnum: true,
+      splitObject: true,
+      splitTuple: true,
+      splitArray: false,
+      splitUnion: true,
+    };
+    return split(model, metaModelsToSplit);
+  }
+
+  constrainToMetaModel(model: MetaModel): ConstrainedMetaModel {
+    // How to constrain Union member names?
+    if (model instanceof UnionModel) {
+      model.union = model.union.map((v, idx) => {
+        v.name += idx;
+        return v;
+      });
+    }
```

Comment:
Yeah, I have a feeling users will want to customize the way Union member names are generated. I couldn't do this in `ModelNameConstrainer.ts` because `ModelNameContext` only passes `modelName` to its fn partial context, but I need the full model to apply this naming constraint. 

In Rust, a Union type is implemented from an `enum` of all possible compositions - so the members need to be named. Here's a concrete example:

```typescript
// The `members` property is a union of one of: string, number, boolean
const doc = {
  $id: '_address',
  type: 'object',
  properties: {
    members: { oneOf: [{ type: 'string' }, { type: 'number' }, { type: 'boolean' }], },
}

// Typescript's expressive semantics allow us to define a union type in-place in the type declaration of Address
// members is composed from: string | number | boolean
class Address {
  private _members?: string | number | boolean;
  constructor(input: {
    members?: string | number | boolean,
  }) {
    this._marriage = input.marriage;
    this._members = input.members;
  }
  get members(): string | number | boolean | undefined { return this._members; }
  set members(members: string | number | boolean | undefined) { this._members = members; }
}
```

Here's the equivalent in Rust, which has much lower-level semantics for expressing that some memory value can be one of many types: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f81defc7eb7e45ecc85f17ebe3a91b60

The important bit is that Rust's union implementation needs a name for each possible type member.
```rust
// Members is a union type of a String, float, or bool value
#[derive(Clone, Debug, PartialEq)]
pub enum Members {
    Members0(String),
    Members1(f64),
    Members2(bool),
}
```

Reply:
That makes sense, thanks for the clarification 👍 Yea, have this on a backburner for other languages cause I dont know the best way to handle them as of yet (https://github.com/asyncapi/modelina/issues/393, https://github.com/asyncapi/modelina/issues/392, https://github.com/asyncapi/modelina/issues/391). So you are the first to have this need.

It's a bit of a conundrum, cause I am not sure `ModelNameConstrainer` make sense to place this logic in 🤔 Even if you had access to the union model in the context and ran this code, it would be overwritten once the constrainer walk over the union models next. So it will be somewhat of a mess 🙃 Need to think about that one...

I think we should keep it here for now and I will make another issue about this as a follow-up, how does that sound?